eradiate/radprops/vert_dist.py

```diff
@@ -0,0 +1,409 @@
+                x = z.magnitude
+                loc = self.mean.to(z.units).magnitude
+                scale = self.std.to(z.units).magnitude
+                f = norm.pdf(x=x, loc=loc, scale=scale)
+                return self._normalise(f)
+            else:
+                raise ValueError(
+                    f"Altitude values do not lie between layer "
+                    f"bottom ({self.bottom}) and top ({self.top}) "
+                    f"altitudes. Got {z}."
+                )
+
+        return eval
+
+
+@VerticalDistributionFactory.register("array")
+@parse_docs
+@attr.s
+class Array(VerticalDistribution):
+    """
+    Flexible vertical distribution specified either by an array of values,
+    or by a :class:`~xarray.DataArray` object.
+    """
+
+    values = documented(
+        attr.ib(
+            default=None,
+            converter=attr.converters.optional(np.array),
+            validator=attr.validators.optional(attr.validators.instance_of(np.ndarray)),
+        ),
+        doc="Particles fractions values on a regular altitude mesh starting "
+        "from the layer bottom and stopping at the layer top altitude.",
+        type="array",
+        default="``None``",
+    )
+    data_array = documented(
+        attr.ib(
+            default=None,
+            converter=attr.converters.optional(xr.DataArray),
+            validator=attr.validators.optional(
+                attr.validators.instance_of(xr.DataArray)
+            ),
+        ),
+        doc="Particles vertical distribution data array. Fraction as a function"
+        " of altitude (``z``).",
+        type=":class:`xarray.DataArray`",
+        default="``None``",
+    )
+
+    @data_array.validator
+    def _validate_data_array(self, attribute, value):
+        if value is not None:
+            if not "z" in value.coords:
+                raise ValueError("Attribute 'data_array' must have a 'z' " "coordinate")
+            else:
+                try:
+                    units = ureg.Unit(value.z.units)
+                    if not units_compatible(units, ureg.Unit("m")):
+                        raise ValueError(
+                            f"Coordinate 'z' of attribute "
+                            f"'data_array' must have units"
+                            f"compatible with m^-1 (got {units})."
+                        )
+                except AttributeError:
+                    raise ValueError(
+                        "Coordinate 'z' of attribute 'data_array' " "must have units."
+                    )
+
+    method = documented(
+        attr.ib(
+            default="linear", converter=str, validator=attr.validators.instance_of(str)
+        ),
+        doc="Method to interpolate the data along the altitude. \n"
+        "This parameter is passed to :meth:`xarray.DataArray.interp`.",
+        type="str",
+        default='``"linear"``',
+    )
+
+    def __attrs_post_init__(self):
```

Comment:
Please stuff that into an `update()` method and make `__attrs_post_init__()` call it.